typegen/evaluation/file_type_hints_collector.py

```diff
@@ -0,0 +1,159 @@
+import os.path
+import pathlib
+from typing import Dict
+import pandas as pd
+import libcst as cst
+from libcst.metadata import PositionProvider
+import constants
+from tracing import TraceDataCategory
+
+
+class FileTypeHintsCollector:
+    """Collects the type hints of multiple .py files."""
+    def __init__(self, project_dir: pathlib.Path):
+        self.project_dir = project_dir
+        self.typehint_data = pd.DataFrame()
+        self.file_path = ""
+
+    def collect_data(self, file_paths: list[pathlib.Path]) -> None:
+        """Collects the type hints of the provided file paths."""
+        for file_path in file_paths:
+            with file_path.open() as file:
+                file_content = file.read()
+
+            module = cst.parse_module(source=file_content)
+            module_and_meta = cst.MetadataWrapper(module)
+            relative_path = str(pathlib.Path(os.path.relpath(file_path.resolve(), self.project_dir.resolve())))
+            visitor = _TypeHintVisitor(relative_path)
+            module_and_meta.visit(visitor)
+            typehint_data = visitor.typehint_data.astype(constants.TraceData.TYPE_HINT_SCHEMA)
+
+            self.typehint_data = pd.concat(
+                [self.typehint_data, typehint_data], ignore_index=True
+            ).astype(constants.TraceData.TYPE_HINT_SCHEMA)
+
+
+class _TypeHintVisitor(cst.CSTVisitor):
+    METADATA_DEPENDENCIES = (PositionProvider,)
+
+    def __init__(self, file_path: str) -> None:
```

Comment:
Which would make this a `pathlib.Path` too

Reply:
There isn't any usage for file path being pathlib.Path. In fact, if it were a path, it would have to be converted to a str.

Reply:
Ah I see, you only use it to write into the DataFrame? That's fine then :+1: